src/module.c

```diff
@@ -9107,6 +9108,19 @@ void RM_SetClusterFlags(RedisModuleCtx *ctx, uint64_t flags) {
         server.cluster_module_flags |= CLUSTER_MODULE_FLAG_NO_REDIRECTION;
 }
 
+/* Returns the cluster slot of a key, similar to the `CLUSTER KEYSLOT` command.
+ * This function works even if cluster mode is not enabled. */
+unsigned int RM_ClusterKeySlot(RedisModuleString *key) {
+    return keyHashSlot(key->ptr, sdslen(key->ptr));
+}
+
+/* Returns a short string that can be used as a key or as a hash tag in a key,
+ * such that the key maps to the given cluster slot. Returns NULL if slot is not
+ * a valid slot. */
+const char *RM_ClusterSampleKeyNameBySlot(unsigned int slot) {
```

Comment:
I don't really like this name, since it also sort of sounds like it's sampling key names. Maybe?
```
const char *RM_ClusterExampleKeyNameInSlot(unsigned int slot) {
```
It's an example key that is in the slow. Also maybe:
```
const char *RM_ClusterDeterminsiticKeyNameInSlot(unsigned int slot) {
```
To make it clear that this will always return the same name for slot? I wonder if we should force that though. I'm trying to think of a reason we might want to change it. We've talked about support for larger slot sizes.

Reply:
Good points. It's useful to rely on it being deterministic. Canonical is another word, `RM_ClusterCanonicalKeyNameInSlot`.

If we don't want to force determinism, then Example is good.

What do you mean by larger slot sizes? Wouldn't that break clients?

Reply:
> RM_ClusterCanonicalKeyNameInSlot

I like canonical. I'm good with this name.

> What do you mean by larger slot sizes? Wouldn't that break clients?

Yes it would, but we at AWS have seen customers start to see issues with the slot limits. It'll likely not become an issue until we start approaching clusters around 4k nodes in size, since we see the hottest slots have about 4x the mount of traffic of the coldest slots just because of probability. So at 4k nodes, even isolating the hottest slot on it's own node would likely not be enough. Anyways, this isn't a hard requirement, just something I would like to be thinking about.